### PR TITLE
Cleanup and test format_instance_name

### PR DESCRIPTION
**Pull request by abrown-sg**
* Moved function out of provisioners and into utilities
* Refactored to be more clear
* Added tests

### CODE REVIEW

**Comment by retr0h** [comment]
Seems logical +1

File: molecule/utilities.py [hunk 1]
```diff
@@ -94,6 +94,37 @@ def write_file(filename, content):
...
+    :param platform: the current molecule platform in use
+    :param instances: the current molecule instances dict in use
+    :return: formatted instance name if found in instances, otherwise None
+    """
+    working_instance = None
+
+    # search instances for given name
+    for instance in instances:
+        if instance['name'] == name:
+            working_instance = instance
```
Comment (rgreinho) [inline molecule/utilities.py]: Could you break as soon as you found a match? or do you want the last one?

**Comment by rgreinho** [comment]
:+1: So much clearer! And the UTs help a lot. Great job!

File: molecule/utilities.py (hunk 1 above)
Comment (abrown-sg) [inline molecule/utilities.py]: Good call. Added a break.

**Pull request merged**